src/components/world/wire/wire.js

```diff
@@ -1,56 +1,60 @@
-import { div } from '@cycle/dom'
+import { div, img } from '@cycle/dom'
 import xs from 'xstream'
-import { ANIMATION_TIMEOUT } from '../../../constants'
+import { ANIMATION_TIMEOUT, STOP_EVENT } from '../../../constants'
 import { addDelay } from '../../../utils'
 
 const STEPS = 15
 const STEP_TIMEOUT = ANIMATION_TIMEOUT / STEPS
-
 const step = (a, i) => (a ? i * 2 : 0)
 const flow = i => (i % 2 === 0 ? 1 : -1)
 const translateX = (a, i) => `translateX(${step(a, i)}vw) translateY(${flow(i)}vh)`
 const translateY = (a, i) => `translateY(${step(a, i)}vh) translateX(${flow(i)}vw)`
+const style = (animate, direction, i) => ({
```

Comment:
`style` could take a `state` argument and you can destructure it.

Then what the real purpose of `direction` into the state if it depends on `type` ?
You can write `direction === 'top'` or `type === 'musics'` with same results